mlx_lm/generate.py

```diff
@@ -363,47 +363,71 @@ def generate_step(
 
     sampler = sampler or (lambda x: mx.argmax(x, axis=-1))
 
-    def _model_call(y):
-        if y.ndim == 3:
-            return model(None, cache=prompt_cache, input_embeddings=y)
+    def _model_call(input_tokens: mx.array, input_embeddings: mx.array | None):
```

Comment:
The `|` is Python 3.10+. So maybe better to drop the type hints here (or use a `Union` if you want type hints)